### DISCUSSION TITLE
Nightmare Project / Development Hell Stories

### DISCUSSION THREAD

**Discussion by WithinAmnesia** [flareteam/flare-game#909] [Q&A]
I know myself too well, I want to do endless work on art and world building; yet I really need to get the itemization and ~'core mechanics' for driving game play sorted out first. I have to be patient and wait for 1.14's large scale itemization features before I go bonkers with non-future-proof itemization system core prototypes / make thousands of soon-to-be-deprecated prototype items. I am hoping to avoid ~'retrofit hell' XD.

I want to make this Castle into a tileset for Flare https://www.deviantart.com/withinamnesia/art/Balduran-s-Sea-Tower-Redesign-W-I-P-17-818093805 + https://www.deviantart.com/withinamnesia/art/The-Bridge-Tower-Gatehouse-W-I-P-02-818683405 + https://3dwarehouse.sketchup.com/collection/u183f210c-3b29-428a-a070-c390ce74e58f/Rochester-Castle?sortBy=createTime%20DESC&searchTab=model&hl=en . Yet I feel that adding art is going 'wide' and right now I need to go 'tall' and create the ~'bones' / itemization / character attribute prototypes of the game first.

I remember not focusing on core mechanical issues of a game engine and putting them off and going into art and items and going 'wide' when I had issues going 'tall'. Yet I hope that I know better now and I should first be tackling the game engine mechanical core issues (this case is 1.14 itemization). I remember in the past getting ~'distracted' / not working on the most important issues first / not prioritizing the ~'core mechanics' of my RPG and Unity related game projects; when I had to be patient / wait / try harder for game engine related work.

I later regretted this ~'impatience' / 'distraction' of not staying focused on the core engine issues from the start. In the past I made the critical mistake of assuming that I can fix the engine issues later after I work on the more 'fun' / easier parts that I liked in the moment. For sometimes I would find out after doing all of / a huge amount of art and 'as is' game prototyping that some key features that I needed in the ~'near-future' were just inherently broken and or were no where practical to implement; even from the start. Let alone being neck deep in potential retrofit work / trying to change thousands of pieces of content to hopefully work with new changes; that might not even work to boot and will absolutely add layers upon layers of bugs XD.

Yet I only found out about the nightmare that is ~'retrofit hell', _after_ years of making the art and items and 'as is' prototyping. When I finally got around and took a deep dive into the engine issues mid-development cycle trying to make the 'near-future' into into the present. I found out that all of my previous work was inherently broken to the core and un-salvageable. For the core engine was more than broken beyond hope and I used non-opensource / proprietary art that I did not own the copyright to as a base for my work which I could not take to a new RPG engine to save my past ~half decade of work. Thus the whole works of the new art assets and the entire project had to be scrapped and killed in its entirely due to ~'retro-fit hell' / endless and exponentially complicated development cycles.

In fact my entire RPG Maker MV game project had to be scrapped / I killed my entire game project after ~4-6 years of work for the engine was unable to grow without endless fractal / near-infinite growth of engine bugs and complexity. It was a true nightmare project / development hell. I turned a 2D game engine 3D and modded tricycle into a helicopter and it still was not enough to fix core engine issues that I should have started on / addressed first / tackled day one.

Here is a video of a late stage prototype of that nightmare project before I just gave up on the nightmare project / development hell: https://youtu.be/r2Bw3xeV1Po?t=240 . I had to let go of everything in order to save my energy / peace of mind / [it was just time to stop](https://www.youtube.com/watch?v=2k0SmqbBIpQ&ab_channel=TooDamnFilthy). I went back into the ~digital wilderness for a fresh hopefully future proofed start / hopefully Flare works lol XD. Do not use RPG Maker _anything_ for making an RPG / half-decent video game XD.

Has anyone else had to experience anything similar like that kind of nightmare project / development hell? As kind of like where one started out and put the core issues off to work on more ~'fun' parts of development and then realize / way down the line / mid-development cycle / _years_ later that the core of the project is mortally flawed beyond repair? That almost sounds like a failed relationship but for a video game development story now that I look at it lol XD.

I suppose that I am wondering what other 'nightmare projects' / development hell stories that the community has experienced? I wonder if anyone is willing to share their stories of development hell or something like that? It doesn't have to be huge or crazy. It can be just anything you think might fit / make sense to share; it is always nice to hear feedback or even a personal story :-).

**Comment by Danimal696** [comment]
Quite roundabout way to ask Dorkster how progress is going, but I agree. I personally am also waiting for the integration of "enemy level up by declaration", "item randomization creator" and "random use of declared body parts".

But you need to take into account that only 2-3 weeks has passed, he has a life to take care of like everyone else and to him this is just a passion project. He has always delivered on his promises to complete internet strangers so far; so we just need to wait around, it´s not like there are not things you can improve without him on your Flare mod.

**Reply by WithinAmnesia** [reply]
@Danimal696 (Note: I wrote a small book of a reply for I felt that a paragraph was incomplete to how I feel / what I actively contend with / try to learn / discover with in life / feel at this moment)
Dorkster should feel free to do what ever he wants on his own time. I do not want to press him for anything at all. I feel that it is already a small miracle the he works for the community for free. I am very grateful if not subconsciously haunted by a sense of guilt in that I cannot / have not give / given as much as he has to the community for free.

I don't care / mind / I am not upset if it takes a year or more for the patch to come out or even longer or never does. It is an odd feeling yes, but the way I look at it is multilayered. Everything that has been done / worked on / developed up to this point is more than I have contributed.

So I do not feel as if I have a leg to stand on to complain nor make demands. Even if everything dies tomorrow I can still make a decent video game. That would maybe not be 100% of what I was aiming at; but still I am not broken down / removed from progressing. So what ever happens, happens.

I can ask for this and that sure, but I have no assumption that anything should / will be produced. Anything given freely is almost like magic / a miracle / comes out of the blue. I do not expect for free work to always continue and be an infinite support. Especially for these admittedly niche development areas / places where so so few people globally have an active interest in / where there is an extremely high turn over / abandon / 'give up' rate / project failure rate.

I do not know anyone here in person. I type, I write books and too long comments yet I wonder what kind of connections can be realistically made in this kind of situation. This kind of social environment feels sort of unnatural or like empty on the inside or that something critical is missing.

I don't think that I can quite fix / improve that observation / feeling / ~'emotional vacuum' by addressing it either. It feels like I have spent the majority of my conscious life in digital places like this but in the end all I find are ghosts and fading memories of thousands of names. Yet it is like trying to reach out and touch the stars in the night sky. It is so empty yet so alive / brilliant / inspiring and ever expansive / impossibly larger than one's own self.

I am not sure what kind of experience that has done to my self and my psyche as a human being. I am not immune to emotion nor do I try to be / I refuse to be numb. Yet in doing so I wonder if I grow ever more distant from the person whom I once was / started out as and I wonder what I am to become of these conscience decisions to keep pressing onwards unyieldingly towards those omni-brilliant stars floating aloft within the sea of the night sky.

Even now I am bouncing thoughts upon whom ever reads this, like a bat's sonar navigating in the darkness. I am not fully sure what I am at this point. What kind of creature of the night I have become and what I am to ultimately be within my fate of fates. Yet I feel so distant as if thousands of words given to thousands of souls bring nothing but silence of camaraderie and isolation of dreams to my heart of hearts. Yet in the dark of the mind and within the silence of the soul one can truly listen beyond the din of life to what one truly loves.

I do care about the community though. Even as doom and gloom as I may sound, and perhaps even jaded / hardened / galvanized in my outlook and or manner in which I give my disposition to the wider public at large; I am very much a champion of hope and optimism. Yet at times it feels like in order to get to the gates of Heaven one must withstand the gate of Hell. I wonder how everyone around myself feels on the inside. I wonder what they battle with and what they have battled with in the past; win, lose or something else. I want to understand what people feel and what they want in life and or what they think life could be. As if they were asked the question how good could life really be if they tried their hardest to live up to their dreams?

I want the best for people yet ironically it may seem like in life people can be at their best when things get the worst. So I wonder who the people are on the inside that I work with, talk with and generally try to socialize with; admittedly at times resulting in various amounts of success and failure. I wonder as to who they are, as in about how they feel on the inside and out and how they hope, cry, laugh and try to improve their lives in general. I know that they and myself are virtually identical in terms of our humanity. So I keep wondering what people are interested in for I want to better understand what people dream about and hope for. So that in some way I can hope to better improve the lives of people around myself. Also I just can't stop working on improving whatever it is that I am around; it is just the nature of who I truly am.

I suppose that I want to show who I am in ways to have people understand who I am in an a honest context. So as to be approachable / inviting to talk with and possibly / hopefully work with and cooperate with. So I have these kind of ~'spirited' / long winded / open-hearted dialogs / comments about what I battle with / how I struggle and what I am aiming at / what my dreams are. To hopefully be like a beacon of hope / a fire in the night / something to fight against the storms of chaos and rains of pessimism. A good deed shines ever the brighter in this weary world. I try to lead by example / try to show that good things can be done and that life can get better. In that even simple steps taken by anyone, weak or strong, rich or poor can eventually succeed. Yet only if one does not give up and keeps in the fight for what they truly love; truly become unyielding. Love has no meaning without sacrifice; yet sacrifice has no meaning without love.

Admittedly I am maybe not the best at these attempts to inspire the world, provide the basic needs of life for all of humanity to flourish, provide homes to all of the youth, schools for every child and retirement security for the old, end poverty, abolish child abuse and eradicate needless suffering to vastly improve humanity. I may be doing myself a disservice to my aspirations by sprouting out doom and gloom via epics of Gilgamesh / Bibles / books / walls of text in the comments sections; but I do hope to be understood at least. Even if it takes an hour to write and 10 minutes to read embarrassingly enough. I just don't want to give up on good of humanity and the positivity of the human spirit found within everyone in any community across all of the world. I feel like a hammer trying to knit a blanket right now trying to explain how I feel and now I realize that I have run out of thread to weave / bash away trying to weave a tapestry made from the threads of my memories of my experiences in living life with myself and everyone and everything.

**Comment by dorkster** [comment]
Hey guys, sorry for the radio silence. I'll be honest, Elden Ring kind of consumed me and my free time :sweat_smile:

Flare is still very much on my mind. I just pushed an [implementation of XP scaling tables](https://github.com/flareteam/flare-engine/commit/549da3b292d0534ed80eafc20b0557f69559f3c6). I'll be looking at tackling the other 1.14 tasks as well. As always, just try to keep expectations tempered.

**Comment by Danimal696** [comment]
I totally dig the lore and graphics of those games, but they are so fu***ing stressing; they are made for a special type of player. I totally feel like this: https://www.youtube.com/watch?v=LSyRf8rAjTs

Nice update by the way.

**Comment by Danimal696** [comment]
Hello guys, I have been busy with work lately so i don´t drop as much; I was wondering, has there been some advances with enemy scaling or item generator?.

**Reply by dorkster** [reply]
Sorry, I haven't been able to get around to any developments there. However, I do have some [new thoughts](https://github.com/flareteam/flare-engine/issues/1815) on the potential implementation of item randomization.

**Comment by Danimal696** [comment]
Hi guys, just wondering what has been going on around here lately, I didn´t log in the whole summer. Has there been some news?

**Reply by dorkster** [reply]
Nothing significant since the [fractional stat value implementation](https://github.com/flareteam/flare-engine/commit/a826e609962334126336fe3c1ef748f68beedae8) in July. I've been occupied with other programming projects, among some non-programming things going on. So I haven't really had the "itch" to work on Flare as of late.

If I have something that I'm working on, I tend to not sit on it for too long. So changes get pushed fairly quick to the master branches of both the [engine](https://github.com/flareteam/flare-engine/commits/master) and the [game](https://github.com/flareteam/flare-game/commits/master) repos. I'd say it's fair to use those commit logs as a barometer for how active the project is.

**Reply by WithinAmnesia** [reply]
@dorkster Any status update on 1.14 / 1.15 / [Arbitrary definition of resources (e.g. HP and MP)](https://github.com/flareteam/flare-engine/issues/1776)? Any approximate testing / patch / release time? Thank you for all of the work it is greatly appreciated. I can't wait to work on Flare with all the proposed new engine, statistical and itemization features and hopefully make a great R.P.G. that can bring joy, fulfillment and wonder to people across the world!

**Reply by dorkster** [reply]
No. Since development has stagnated to this point, I think it's time to just tag the current state as 1.14 and release. It's only fair to the other contributors that have submitted code and translation updates. They should get to see their work in a stable version without being beholden to my (lack of) desire to work on the engine.

Plus, we've had releases with much smaller change logs in the past. So there's definitely enough here to warrant a new release.

**Reply by WithinAmnesia** [reply]
Oh okay, thank you @dorkster . Yeah I feel you, it's understandable; these things can come in waves of motivation with ups and downs. Although just hang in there for a bit longer and we should have great games come from engine and make everyone proud. That's cool to see 1.14 release soon, you put a lot of really cool features in it too. I really appreciate the fractional values and floats and XP scaling and all the stat and itemization work that makes a huge world of possibilities open up and adds tons of content and progression depth and also appreciate the reverse equipment cycling and the tabs UI work. It looks like 1.15 will be pretty cool too as well. Mate having the arbitrary definition of resources would be incredible. There would be a whole world of possibilities that open up with that. It's like the last really big big thing I think I could ask for and want, all the other stuff like random item generation and socketed / enchant-able items are cool but they are like icing on the cake and arbitrary definition of resources is like another layer of cake and its cream filled!

**Comment by Danimal696** [comment]
Hello Dorkster, my mod is getting forgotten in the OGA forum; I invested a lot of work into it and I think it has enought content and playtime; I would hate to see it sink into the digital void.  Could you please upload it to the mod section of Flare webpage?

https://www.dropbox.com/s/wrr11qsnihajtzq/HERESY%20only%20mod.7z?dl=0

You can use this pic for the thumbnail:
https://opengameart.org/sites/default/files/forum-attachments/triston.jpg

Thanks.

**Reply by dorkster** [reply]
Sure. It was listed on the "Other Mods" page, but I agree that it's substantial enough to get its own page.
